File: StudentAdmissionPortal/Models/Student.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace StudentAdmissionPortal.Models
{
    public class Student
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [StringLength(50)]
        public string FirstName { get; set; }

        [Required]
        [StringLength(50)]
        public string LastName { get; set; }

        [Required]
        
        public string DateOfBirth { get; set; }

        
        public int? NationalityId { get; set; }

        [ForeignKey("NationalityId")]
        public Nationality Nationality { get; set; }

        //public bool IsApproved { get; set; }


        public ICollection<FamilyMembers> FamilyMembers { get; set; }


    }
}

File: student-app/src/models/Student.ts
export interface Student {
    id: number,
    firstName: string,
    lastName: string,
    dateOfBirth: Date
    nationalityId?: number
}

export interface Family extends Student{
    relationshipId: number
    relationship: string
}

export interface Nationality {
    id: number,
    country: string
}